src/main/java/org/skriptlang/skript/elements/expressions/ExprRotateQuaternion.java

```diff
@@ -0,0 +1,126 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package org.skriptlang.skript.elements.expressions;
+
+import org.bukkit.event.Event;
+import org.bukkit.util.Vector;
+import org.eclipse.jdt.annotation.Nullable;
+import org.joml.Quaternionf;
+import org.joml.Vector3f;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.util.Kleenean;
+
+@Name("Rotate Quaternion")
+@Description({
+	"Rotates a Quaternion around an axis or a vector and by a set amount of degrees.",
+	"When you use a vector, the values will be used to rotate the x, y and z."
+})
+@Examples("rotate {_quaternion} around vector from 1, 0, 0 at player by 90 degrees")
+@Since("INSERT VERSION")
+public class ExprRotateQuaternion extends SimpleExpression<Quaternionf> {
+
+	static {
+		if (Skript.isRunningMinecraft(1, 19, 4))
+			Skript.registerExpression(ExprRotateQuaternion.class, Quaternionf.class, ExpressionType.SIMPLE,
+					"%quaternions% rotated (around|on) [the] (:x|:y|:z)-axis by %number% [degrees|radians]",
```

Comment:
Unresolved, quaternions should be added to the vector rotate effect.

Reply:
A quaternion has 4 values. You cannot rotate a quaternion by a quaternion. See https://joml-ci.github.io/JOML/apidocs/org/joml/Quaternionf.html for available methods. You mean as in supporting the destination?

Reply:
You can! Multiplying two quaternions applies the rotation of one to the other.

Reply:
Perhaps it might be better worded as "adding" or "compounding" the quaternions.

Either way, that wasn't actually what i was talking about. I meant using quaternions to rotate vectors. See the transform methods.